### PR TITLE
Add aws-auth v0.3.5

### PR DESCRIPTION
**Pull request by eytan-avisror**

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

**Comment by ahmetb** [comment]
Have you read the naming guidelines? That name won't work.

**Comment by eytan-avisror** [comment]
@ahmetb I have, can you be a bit more specific on why it won't work? is it too generic?

**Comment by ahmetb** [comment]
Has vendor prefix and is too generic. AWS or someone else can submit a plugin with the same name that works entirely differently.

**Comment by eytan-avisror** [comment]
@ahmetb can you suggest a better name for the plugin? the purpose is easy modification/management of the 'aws-auth' configmap present on EKS clusters.
is `aws-auth-manager` any better?

**Comment by eytan-avisror** [comment]
BTW @ahmetb 
Not sure if it makes a difference but this tool is referenced by AWS best practices here: https://github.com/aws/aws-eks-best-practices/blob/master/content/security/docs/iam.md#use-tools-to-make-changes-to-the-aws-auth-configmap

And also the suggestion to add a krew plugin was by AWS here:
https://github.com/keikoproj/aws-auth/issues/30

**Comment by eytan-avisror** [comment]
@ahmetb `aws-eks-auth-mapper` kinda long, but perhaps more specific. would that be more acceptable?

**Comment by ahmetb** [comment]
edit-aws-auth can work too.
aws-auth can work if you can get thumbs up from AWS folks that they don't plan on using the name. I thought you might have a chance there since they're aware of this tool.

**Comment by ahmetb** [comment]
Also the fact that there's a notable existing user community and recognition from the vendor, we can consider this as an established solution and assign the name I think.

**Comment by eytan-avisror** [comment]
Thanks @ahmetb 
I've pinged the original requestor for the plugin (from AWS) to comment here

**Comment by eytan-avisror** [comment]
Thanks @jicowan 
@ahmetb I can make the changes, is it OK though that the plugin name differs from the repo name & binary name?

**Comment by ahmetb** [comment]
Yeah, just make sure the -h/-help message doesn't confuse users with different names.

**Comment by jicowan** [comment]
I would prefer to keep it the same (aws-auth) but if we think it'll be too confusing I'm in favor of changing it.

**Comment by eytan-avisror** [comment]
@jicowan yeah I think it would actually be much nicer as a shorter name i.e. `kubectl aws-auth ...` - given that this is a pretty basic term on EKS clusters I don't see a reason where there might be a conflict with some other tooling.

**Comment by eytan-avisror** [comment]
I think according to @ahmetb it's not about it being 'confusing', but more about if you foresee any conflict with the name in the future e.g. AWS wanting to introduce a different tool with that name as a krew plugin being is the word `auth` is somewhat generic

**Comment by jicowan** [comment]
Ah, we have no intention of releasing a new CLI to modify the aws-auth ConfigMap. Rather, we are looking at ways we can assign Cluster access directly through IAM.

**Comment by ahmetb** [comment]
Fair, then we can assign the name aws-auth especially given the tool is already known, used by a nontrivial amount of users and endorsed.

**Comment by eytan-avisror** [comment]
Thanks @ahmetb & @jicowan 
Please let me know if there is anything else needed in order to get this merged.

**Review by ahmetb — CHANGES_REQUESTED** [review]

File: plugins/aws-auth.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+
```
[the remark this answers is not in the document]
Comment (eytan-avisror) [reply plugins/aws-auth.yaml]: done

File: plugins/aws-auth.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  name: aws-auth
+spec:
+  version: v0.3.5
+  homepage: https://github.com/keikoproj/aws-auth
+  shortDescription: Makes the management of the aws-auth config map for EKS Kubernetes clusters easier
```
[the remark this answers is not in the document]
Comment (eytan-avisror) [reply plugins/aws-auth.yaml]: done

File: plugins/aws-auth.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  description: |
+    Useful for automation purposes, any workflow that needs
```
[the remark this answers is not in the document]
Comment (eytan-avisror) [reply plugins/aws-auth.yaml]: Done

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!


Also, please consider setting up [Krew release automation](https://krew.sigs.k8s.io/docs/developer-guide/release/automating-updates/) which helps you skip manually making updates to your Krew manifests on each new version and send a pull request. It’s a GitHub workflow bot that can run every time you push a new tag. Those PRs are automatically approved.

**Comment by eytan-avisror** [comment]
Thanks @ahmetb !

**Pull request merged**